### PR TITLE
Update SPAdes module to official nf-core and version bump 

### PR DESCRIPTION
**Pull request by jfy133**
TODO: 

- [x] Check output same (and if diff, update output.md)
    - [x] waiting to fix teh warnings.log first: https://github.com/nf-core/modules/pull/6615 
    - [x] Run four tests, check difference
    - [x] Debug ancient_dna profile
- [x] replace the fix_cpus trick

:warning: JAMES: MAKE SURE TO CHECK AFTER TEMPLATE MERGE TO DOUBLE CHECK THE CHEKC_MEGAHIT_CPUS ETC FUNCTIONS :warning: 

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/mag/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/mag _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by jfy133** [comment]
Added the GUNZIP module actually does somthing similar to pre-assemblies, but then local MEGAHIT module also exports uncompressed files... so I guess time to update this too

**Comment by jfy133** [comment]
@nf-core-bot fix linting

**Comment by jfy133** [comment]
Final thing: reimplement the CPU fixing

File: conf/base.config [hunk 1]
```diff
@@ -24,150 +24,148 @@ process {
...
+        cpus   = { check_max(10 * task.attempt, 'cpus') }
+        memory = { check_max(128.GB * task.attempt, 'memory') }
+        time   = { check_max(12.h * task.attempt, 'time') }
     }
     //MEGAHIT returns exit code 250 when running out of memory
     withName: MEGAHIT {
-        cpus          = { check_megahit_cpus (8, task.attempt  ) }
-        memory        = { check_max (40.GB * task.attempt, 'memory' ) }
-        time          = { check_max (16.h  * task.attempt, 'time'   ) }
+        cpus          = { params.megahit_fix_cpu_1 ? 1 : check_max(8 * task.attempt, 'cpus') }
```
Comment (skrakau) [inline conf/base.config:123]: Does that work? I remember vaguely that it didn't work to access `params` here, that's why I think I shifted the corresponding functionality to `nextflow.config` ... but maybe that changed?

**Review by skrakau — COMMENTED** [review]
Great to see :D

File: conf/base.config (hunk 1 above)
Comment (jfy133) [reply conf/base.config:123]: It should do if it's in a closure `{ }`. That's why e.g. `task.attempt` works.

Comment (skrakau) [reply conf/base.config:123]: Did you test?

**Review by skrakau — APPROVED** [review]

File: conf/base.config (hunk 1 above)
Comment (jfy133) [reply conf/base.config:123]: Hrm wait, this can't work as `check_max` should be gone...

**Comment by jfy133** [comment]
For future reference, it does work! In the `base.config`  I set the memory of SPADES and MEGAHIT to 7.MB to force retries, and then fix the CPUs :

```
nextflow run ../main.nf -profile test,docker --outdir ./results --spades_fix_cpus 2 --megahit_fix_cpu_1
```

image

image


When fixing the CPUs, as above, you see in the screenshots above that memory retries but CPUs stay at the values fixed by the parameters (the CPUs in the test.config is 12).

Without specifying the fixed CPUs, retrying indeed increases the CPUs

```bash
nextflow run ../main.nf -profile test,docker --outdir ./results
```

image

**Pull request merged**